packages/fastify/src/withClerkMiddleware.ts

```diff
@@ -63,3 +67,45 @@ export const withClerkMiddleware = (options: ClerkFastifyOptions) => {
     req.auth = requestState.toAuth();
   };
 };
+
+const getAbsoluteUrlFromHeaders = (url: string, headers: Headers): URL => {
+  const forwardedProto = headers.get(constants.Headers.ForwardedProto);
+  const forwardedPort = headers.get(constants.Headers.ForwardedPort);
+  const forwardedHost = headers.get(constants.Headers.ForwardedHost);
+
+  const fwdProto = getFirstValueFromHeaderValue(forwardedProto);
+  let fwdPort = getFirstValueFromHeaderValue(forwardedPort);
+
+  // If forwardedPort mismatch with forwardedProto determine forwardedPort
+  // from forwardedProto as fallback (if exists)
+  // This check fixes the Railway App issue
+  const fwdProtoHasMoreValuesThanFwdPorts =
+    (forwardedProto || '').split(',').length > (forwardedPort || '').split(',').length;
+  if (fwdProto && fwdProtoHasMoreValuesThanFwdPorts) {
+    fwdPort = getPortFromProtocol(fwdProto);
+  }
+
+  try {
+    return new URL(url, `${fwdProto}://${forwardedHost}${fwdPort ? ':' + fwdPort : ''}`);
+  } catch (e) {
+    return new URL(url, DUMMY_URL_BASE);
+  }
+};
+
+const PROTOCOL_TO_PORT_MAPPING: Record<string, string> = {
+  http: '80',
+  https: '443',
+} as const;
+
+function getPortFromProtocol(protocol: string) {
+  return PROTOCOL_TO_PORT_MAPPING[protocol];
+}
+
+function getFirstValueFromHeaderValue(value?: string | null) {
+  return value?.split(',')[0]?.trim() || '';
+}
+
+const isRelativeUrl = (url: string) => {
```

Comment:
🙃 let's add a comment for this one to move it to `shared` package or somewhere else and re-use it from there. I believe we have that code somewhere too.